src/mawpy/utilities/validations.py

```diff
@@ -0,0 +1,12 @@
+from mawpy.constants import INPUT_ARGS_LIST
+
+
+def check_non_negative(value: float) -> None:
+    if value < 0:
+        raise ValueError(f"{value} is not a valid non-negative numerical value")
+
+
+def validate_input_args(**kwargs: float) -> None:
```

Comment:
Since this only validates non-negative values, I suggest renaming it to `validate_non_negative_args` and then just inlining the `<0`. This is simpler and clearer. 

But if we expect this will need to generalize further you could create a map of validation functions for each arg, eg:
```
validations = { 
   "duration": [check_non_negative, check_finite]
   "spatial_constraint: [partial(check_range, min=0.1, max=10.0)], 
}
```

Reply:
Updated as per suggestion.